[Xing Wei] iP

Moley456 opened the pull request:
## Duke
> Your mind is for having ideas, not holding them. – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Duke helps in managing things you need to do, making it much less taxing to remember each of your tasks!
Duke is,
* Text-based
* Easy to learn
* Fast to use

To use Duke just,
1. Download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html)
2. Double click it
3. Add your tasks
4. Relax knowing that you will never forget your tasks again! :blush:

And, it is **FREE** :grin:

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }
```

Commit:
Add Gradle support

Moley456 committed:
implement Duke level 1

Moley456 committed:
Implement Level 2 Duke - Add and List tasks

Moley456 committed:
Implement Level 3 Duke - Mark as Done

Moley456 committed:
Implement Level 4 Duke - ToDo, Event, Deadline

Moley456 committed:
Update Automated Text UI Testing inputs and expected outputs

Moley456 committed:
Fix SWE practices and conventions

Moley456 committed:
Implement Level 5 Duke - Handle Exceptions

Moley456 committed:
Update inputs and expected output for Automated Text UI Testing

Moley456 committed:
Implement Level 6 Duke - Delete

Moley456 committed:
Update input and expected output for Automated Text UI Testing

Moley456 committed:
Implement Duke Level 7 - Save

Moley456 committed:
Implement Duke Level 8 - Dates

Moley456 committed:
Revert "Implement Duke Level 8 - Dates"

This reverts commit ddb78e60ff8fcf9fed8b6f74bb6445de07564974.

Moley456 committed:
Implement Duke Level 8 - Dates

Moley456 committed:
Merge branch 'branch-level-7'

Moley456 committed:
Merge branch 'branch-level-8'

Moley456 committed:
Minor format fix

Moley456 committed:
Revert "Merge branch 'branch-level-8'"

This reverts commit d304fbc588a683c4dd85b4b58570eba9a51eb9ef, reversing
changes made to deceef47d08fa4ba34d769c9efa4749d8db44252.

Moley456 committed:
Implement Duke Level 7 - Save

Moley456 committed:
Merge branch 'branch-level-7'

Moley456 committed:
minor text change

Moley456 committed:
Merge branch 'branch-level-8'

Moley456 committed:
minor code formatting

Moley456 committed:
Fix Duke Level-8

Moley456 committed:
git fix attempt

Moley456 committed:
Merge branch 'master' of https://github.com/Moley456/ip

# Conflicts:
#	src/main/java/Deadline.java

Moley456 committed:
Fix Duke support for dates

Moley456 committed:
Merge branch 'branch-level-8' of https://github.com/Moley456/ip into branch-level-8

Moley456 committed:
Use More OOP

Moley456 committed:
Minor bugfix and handle some exceptions

Moley456 committed:
Package all classes

Moley456 committed:
Add support for time input for Deadlines

Moley456 committed:
Add JUnit tests

Moley456 committed:
Minor format fix

Moley456 committed:
Add JAR file

Moley456 committed:
Add Javadoc comments

Moley456 committed:
Fix coding standards

Moley456 committed:
Implement Duke Level 9 - Find

Moley456 committed:
Merge branch 'branch-A-JavaDoc'

Moley456 committed:
Merge branch 'branch-A-CodingStandard'

Moley456 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/task/TaskList.java

Moley456 committed:
Update "find" functionality to be case insensitive

Moley456 committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/ToDo.java

Moley456 committed:
Remove branch-level-8 files after late merge to master

srj31 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+
+        Scanner sc = new Scanner(System.in);
+        Parser parser = new Parser(tasks);
+        boolean end = false;
```
based on the coding standard, shouldn't this be isEnd? (i.e. a boolean should have a boolean name)

srj31 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+     *
+     * @return The letter "D" that represents a Deadline.
+     */
+    public String taskType() {
```
Perhaps could have used getTaskype() for other Tasks as well

DesmondTo replied:
Agree with srj31 :)

srj31 reviewed (commented):
Coding Standards have been followed well and the code is reader friendly as well.

Moley456 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Moley456 committed:
Update build.gradle

Moley456 committed:
Add checkstyle and fix checkstyle code violations

Moley456 committed:
Partial transition from CLI to GUI

Moley456 committed:
Implement full Duke functionality with GUI

Moley456 committed:
Design the GUI, add/change javadocs, change JUnits tests

Moley456 committed:
Merge branch 'branch-level-10'

DesmondTo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,89 @@
+package duke;
+
```
I think there's one redundant line here :)

DesmondTo commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,35 @@
+import javafx.stage.Stage;
+
+
+
```
May be it's better to remove some gaps here :)

DesmondTo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,121 @@
+                case "E":
+                    loadedTaskList.add(new Event(lineSplit[2], lineSplit[3]));
+                    break;
+                default:
```
If the default did nothing, I think it is better not to put it :)

DesmondTo commented on src/main/java/duke/controllers/MainWindow.java:
```diff
@@ -0,0 +1,61 @@
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+
```
Again I think it's good to remove some gaps here haha

DesmondTo commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+     *
+     * @return The letter "E" that represents an Event.
+     */
+    public String taskType() {
```
Name it as `getTaskType` sounds better

DesmondTo commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,149 @@
+
+        for (int i = 0; i < tasks.size(); i++) {
+            String currentTask = tasks.get(i).toString();
+            if (currentTask.substring(7).matches("(?i)(.*)" + keyword + "(.*)")) {
```
I think you can improve this line by assign this line: `currentTask.substring(7).matches("(?i)(.*)" + keyword + "(.*)")` into a descriptive boolean variable so it is easier tor read :)

DesmondTo reviewed (commented):
There are some nits to fix but most of the codes follow the expected standards, well done!!

Moley456 committed:
Update build.gradle and README

Moley456 committed:
Add assertions

Moley456 committed:
Improve code quality for better readability and reusability.

Large chunks of code in methods.
Some String literals are duplicated in the code.

Large chunks of code hinder readability and it is better not to have
duplicated code.

Refactoring the methods with large number of lines into multiple shorter
methods will help to improve readability and will improve ease of
changing the code in the future.

Moley456 committed:
Merge pull request #2 from Moley456/branch-A-CodeQuality

Improve code quality for better readability and reusability.

Moley456 committed:
Merge branch 'master' into branch-A-Assertions

Moley456 committed:
Merge pull request #3 from Moley456/branch-A-Assertions

Add assertions

Moley456 committed:
Merge branch 'master' into branch-A-Assertions

Moley456 committed:
Fix naming of Boolean variable

Moley456 committed:
Implement Update extension feature

Users are not able to update tasks after they have added.

This becomes quite a hassle as users must delete and add tasks again
to perform an update.

Implementing a feature to allow easier updates of tasks can improve
ease of use for the users. Users will be able to update either the
task's description, date & time or the entire task.

Moley456 committed:
Add and Update Javadocs

Moley456 committed:
Merge branch 'branch-C-Update'

Moley456 committed:
Merge branch 'branch-A-CodeQuality'

Moley456 committed:
Allow events to handle date and time properly and clean up code.

Moley456 committed:
Change implementation such that bye command closes Duke application

Moley456 committed:
Improve Duke GUI

Moley456 committed:
Update README.md

Moley456 committed:
Update user guide

Moley456 committed:
Update user guide

Moley456 committed:
Fix minor bugs and upload images

Moley456 committed:
Update user guide format

Moley456 committed:
Set theme jekyll-theme-cayman

Moley456 committed:
Update user guide format

Moley456 committed:
Update user guide

Moley456 committed:
Update README.md

Moley456 committed:
Update user guide

Moley456 committed:
Update README.md

Moley456 committed:
Update user guide
